### PR TITLE
[release2_1] [whl] fix stability problem: failed to boot up SOS in cold boot  

### PR DESCRIPTION
**Pull request by conghuic23**
When wake_thread is called right after sleep_thread on the same vcpu,
but be_blocking is still true, which means the block progress is not
finished, then the thread status would become BLOCKED forever.

To fix the issue, during wake_thread, if be_blocking is true, also go
through the wake flow, the thread is put back to runqueue, and the
status is changed to RUNNABLE

But this is not enough, as when there is only one thread obj in
runqueue, prev would equal to next in scheudle(), so no context switch
happens, the status cannot changed to RUNNABLE. To fix this, also update
the status for case prev == next.

Tracked-On: #5122
Signed-off-by: Conghui Chen <conghui.chen@intel.com>

### CODE REVIEW

**Comment by conghuic23** [comment]
@shuox  please also help to review this patch.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/common/schedule.c [hunk 1]
```diff
@@ -183,6 +183,10 @@ void schedule(void)
 		release_schedule_lock(pcpu_id, rflag);
 		arch_switch_to(&prev->host_sp, &next->host_sp);
 	} else {
+		/* there is a case, when prev == next, and next's
+		 * status is runnable, then, we should make sure
+		 * the status is running after schedule.*/
```
Comment (shuox) [inline hypervisor/common/schedule.c:188]: Do you mean prev==next and its status is RUNNABLE? How a RUNNABLE thread execute schedule()?

**Comment by fyin1** [comment]
Let's wait for Shuo's comment addressed.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by shuox** [comment]
@fyin1 patch updated.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**